### ISSUE TITLE
Pattern for redirect plugin should include optional link to section heading

### ISSUE THREAD

**Issue by jfairchild** [squidfunk/mkdocs-material#4690]
### Contribution guidelines

- [X] I've read the [contribution guidelines](https://github.com/squidfunk/mkdocs-material/blob/master/CONTRIBUTING.md) and wholeheartedly agree

### I've found a bug and checked that ...

- [X] ... the problem doesn't occur with the `mkdocs` or `readthedocs` themes
- [X] ... the problem persists when all overrides are removed, i.e. `custom_dir`, `extra_javascript` and `extra_css`
- [X] ... the documentation does not mention anything about my problem
- [X] ... there are no open or closed issues that are related to my problem

### Description

https://github.com/squidfunk/mkdocs-material/blob/master/docs/schema/plugins/external/redirects.json
The Pattern for Internal redirects is `"\\.md$"` however, the redirect plugin added support for redirects to section headings.

PR to the plugin repo: https://github.com/mkdocs/mkdocs-redirects/pull/42

I'm opening the issue now, but will eventually open a PR to update the schema json.

I think something like `\\.md(#\S+)?$` should work.

### Expected behaviour

The yaml schema check should allow linking to section headers

### Actual behaviour

<img width="568" alt="image" src="https://user-images.githubusercontent.com/1012652/203653027-92db30a9-1595-47b2-88ec-7606a187e232.png">
The `yaml-schema` check fails when linking to a section heading.

Example config:
```yaml
plugins:
  - search
  - redirects:
      # https://pypi.org/project/mkdocs-redirects/
      redirect_maps:
        'standards/old.md': 'standards/new.md#new-standards'
```

### Steps to reproduce

Tested on docker image version: squidfunk/mkdocs-material:8.5.10

`docker run ... squidfunk/mkdocs-material:8.5.10`
`docker run ... squidfunk/mkdocs-material:8.5.10 build --strict -v`

Example config:
```yaml
plugins:
  - search
  - redirects:
      # https://pypi.org/project/mkdocs-redirects/
      redirect_maps:
        'standards/old.md': 'standards/new.md#new-standards'
```

when navigating to `/standards/old/` you'll get redirected to `/standards/new/#new-standards`

### Package versions

- Python: `python --version`
- MkDocs: `mkdocs --version`
- Material: `pip show mkdocs-material | grep -E ^Version`

### Configuration

```yaml
plugins:
  - search
  - redirects:
      # https://pypi.org/project/mkdocs-redirects/
      redirect_maps:
        'standards/old.md': 'standards/new.md#new-standards'
```

### System information

Client:
 Cloud integration: v1.0.25
 Version:           20.10.16
 API version:       1.41
 Go version:        go1.17.10
 Git commit:        aa7e414
 Built:             Thu May 12 09:20:34 2022
 OS/Arch:           darwin/amd64
 Context:           default
 Experimental:      true

Server: Docker Desktop 4.9.0 (80466)
 Engine:
  Version:          20.10.16
  API version:      1.41 (minimum version 1.12)
  Go version:       go1.17.10
  Git commit:       f756502
  Built:            Thu May 12 09:15:42 2022
  OS/Arch:          linux/amd64
  Experimental:     false
 containerd:
  Version:          1.6.4
  GitCommit:        212e8b6fa2f44b9c21b2798135fc6fb7c53efc16
 runc:
  Version:          1.1.1
  GitCommit:        v1.1.1-0-g52de29d
 docker-init:
  Version:          0.19.0
  GitCommit:        de40ad0

**Comment by squidfunk** [comment]
Thanks for reporting. Awesome to see that new functionality, which we should definitely support in our schema!

**Comment by squidfunk** [comment]
Closed by #4691.
